app/models/concerns/client_search.rb

```diff
@@ -35,7 +35,18 @@ def self.text_searcher(text, sorted:, resolve_for_join_query: false)
         where = sa[:DOB].eq("#{year}-#{month}-#{day}")
       elsif numeric
         where = sa[:PersonalID].eq(text)
-        where = where.or(sa[:id].eq(text)) if term_is_possibly_pk
+        if term_is_possibly_pk
+          conditions = [where, sa[:id].eq(text)]
+
+          # if we are dealing with HMIS client records, match against deleted/merged ids
+          if self == Hmis::Hud::Client
```

Comment:
@eanders should the warehouse also support searching by Client IDs that have been deleted due to a merge in HMIS? I don't see why not.

Reply:
no harm in it